sdk/src/Core/Amazon.Runtime/Telemetry/Tracing/TracerProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+namespace Amazon.Runtime.Telemetry.Tracing
+{
+    /// <summary>
+    /// Abstract class representing a provider for creating tracer instances.
+    /// Tracer providers are used to create tracers which record tracing data.
+    /// </summary>
+    public abstract class TracerProvider
+    {
+        /// <summary>
+        /// Creates a new <see cref="Tracer"/> instance for a given scope.
+        /// Tracers are used as an entry point for creating spans.
+        /// </summary>
+        /// <param name="scope">The name of the instrumentation scope that uniquely identifies the tracer.</param>
+        /// <returns>A <see cref="Tracer"/> instance for the specified scope.</returns>
+        public abstract Tracer GetTracer(string scope);
```

Comment:
The SRA contained an optional attributes for `getTracer`, however `OTel.GetTracer` doesn't accept attributes as a parameter.
I checked c++ implementation and they take this parameter and ignore it when for the OTel wrapper 
![image](https://github.com/aws/aws-sdk-net-staging/assets/28155926/041de6ec-bd73-4082-9a72-4230a5978dc9)
And Kotlin only allows `scope` for this API 
![image](https://github.com/aws/aws-sdk-net-staging/assets/28155926/f54681c1-615c-4177-80d3-7ea8b66f91b0)

I chose to only allow `scope` rather than allowing an additional attributes parameter that we will be ignored by OTel implementation.